### PR TITLE
New insert and some stuff.

### PR DESCRIPTION
**Pull request by lars-timezynk**
There are two `insert` functions now, one that throws exception on nil, one that punts nil.

### CODE REVIEW

File: src/com/timezynk/mongo.clj [hunk 1]
```diff
@@ -404,13 +411,25 @@
...
+   (count-method (coll/get-collection coll)
+                 (convert/clj->doc query))))
 
 ; ------------------------
 ; Insert
 ; ------------------------
 
+(defn- ^:no-doc insert-docs [coll doc options]
+  {:pre [coll]}
+  (if (= [] doc) ; the empty list should not throw exception
```
Comment (eureton) [inline src/com/timezynk/mongo.clj]: Why not call `empty?` here?

File: src/com/timezynk/mongo.clj [hunk 2]
```diff
@@ -404,13 +411,25 @@
...
+  ([coll query]
+   {:pre [coll query]}
+   (count-method (coll/get-collection coll)
+                 (convert/clj->doc query))))
 
 ; ------------------------
 ; Insert
 ; ------------------------
 
+(defn- ^:no-doc insert-docs [coll doc options]
```
Comment (eureton) [inline src/com/timezynk/mongo.clj]: Naming second parameter `doc` is confusing. Please use plural.

File: src/com/timezynk/mongo.clj [hunk 3]
```diff
@@ -441,11 +460,13 @@
...
   [coll doc & options]
-  (let [doc (convert/clj->doc doc)]
-    (-> (coll/get-collection coll)
-        (insert-options options)
-        (insert-method doc))
-    (convert/doc->clj doc)))
+  (insert-docs coll doc options))
+
+(defn insert
+  "This is identical to `insert!`, except if payload is nil, do a nil-punt instead of throwing exception."
```
Comment (eureton) [inline src/com/timezynk/mongo.clj]: It is _pun_, not _punt_ (see [here](https://ericnormand.me/podcast/what-is-nil-punning)).

To pass `nil` through is _"to pun nil"_.

File: test/com/timezynk/mongo/test/fetch_and_replace.clj [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+(deftest replace-and-fetch-old
+  (m/insert! :coll {:name "1"})
+  (is (= {:name "1"}
+         (-> (m/fetch-and-replace-one! :coll
+                                       {}
+                                       {:mail "2"})
+             (dissoc :_id))))
+  (is (= {:mail "2"}
+         (-> (m/fetch-one :coll)
+             (dissoc :_id))))
```
Comment (eureton) [inline test/com/timezynk/mongo/test/fetch_and_replace.clj]: This asserts two things:
1. the document was indeed replaced
1. no other document is present in the collection

The latter looks out of place - unintended perhaps?

Comment (eureton) [reply test/com/timezynk/mongo/test/fetch_and_replace.clj]: I only comment here, but consider the same to apply to the rest of the tests in this PR.

**Review by eureton — DISMISSED** [review]
I approve in advance, but take a look the comments, please.

File: src/com/timezynk/mongo.clj (hunk 1 above)
Comment (lars-timezynk) [reply src/com/timezynk/mongo.clj]: Because if doc isn't a list or nil, `empty?` throws an exception.

**Pull request merged**